File: easy/python/Question121.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        pre_min = prices[0]
        ans = 0
        for x in prices[1:]:
            if x - pre_min > ans:
                ans = x - pre_min
            if x < pre_min:
                pre_min = x
        return ans


print(Solution().maxProfit([7, 6, 4, 3, 1]))

File: cpp/easy/Question121.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int ans = 0;
        int min_price = prices[0];
        for (int i = 1; i < prices.size(); i++) {
            ans = max(ans, prices[i] - min_price);
            min_price = min(min_price, prices[i]);
        }
        return ans;
    }
};
